### ISSUE TITLE
Unable to add Pipeline to AddChatClient

### ISSUE THREAD

**Issue by darena-pjindal** [SteveSandersonMS/dotnet-ai-workshop#1]
It seems that the latest version of the Microsoft.Extensions.Ai doesn't support the pipeline anymore? I added the preview package to a new project, but I can't add the pipeline features outlined in the tutorials. Is there a particular version of the package that I should be using? I started a new project with all the latest versions but I had to go back to the older version based on the workshop to make it work. 
![image](https://github.com/user-attachments/assets/10a9240a-6d5e-435e-8d32-67f00eab50e1)

**Comment by darena-pjindal** [comment]
Hello @SteveSandersonMS, Thank you so much for the detailed workshop. This is very helpful. However, it seems that the code does not build after updating the packages to the latest version. See the screenshots below. I would appreciate any advice on how to fix this. 
![image](https://github.com/user-attachments/assets/95d270af-c109-4e47-b7c1-991eecd80848)

**Comment by snympi** [comment]
I can confirm I have the same issue @SteveSandersonMS. Great workshop and presentation at .net Conf, but it is hard to follow along if the code is not available or working as in the demo. These are all bits still in preview, but this repo has become my frame of reference and it is hard if we can't repro what you have shown.

**Comment by SteveSandersonMS** [comment]
Thanks for the feedback. I'll be updating this content in January.

In the meantime the code changes you'll need to make are fairly trivial. The APIs have been simplified from:

```cs
builder.AddChatClient(pipeline => pipeline
    .UseMiddlewareA()
    .UseMiddlewareB()
    .Use(innerClient));
```

... to this:

```cs
builder.AddChatClient(innerClient)
    .UseMiddlewareA()
    .UseMiddlewareB();
```

For example:

```cs
hostBuilder.Services.AddChatClient(innerChatClient)
    .UseFunctionInvocation();
```

**Comment by darena-pjindal** [comment]
Thanks so much, @SteveSandersonMS . Happy Holidays.

**Comment by snympi** [comment]
Thanks @SteveSandersonMS - enjoy the break.
